enos/enos.py

```diff
@@ -83,82 +81,53 @@
-        logging.info("Symlinked %s to %s" % (env['resultdir'], link))
-    except OSError:
-        # An harmless error can occur due to a race condition when multiple
-        # regions are simultaneously deployed
-        logging.warning("Symlink %s to %s failed" %
-                (env['resultdir'], link))
-        pass
+    # Generate or get the directory for results
+    env['resultdir'] = _set_resultdir(kwargs['--env'])
+    logging.info("Directory for experiment results is %s", env['resultdir'])
 
     # Loads the configuration file
-    config_file = kwargs['-f']
+    config_file = os.path.abspath(kwargs['-f'])
     if os.path.isfile(config_file):
         env['config_file'] = config_file
         with open(config_file, 'r') as f:
             env['config'].update(yaml.load(f))
-            logging.info("Reloaded config %s", env['config'])
+            logging.info("Reloaded configuration file %s", env['config_file'])
+            logging.debug("Configuratin is %s", env['config'])
     else:
-        logging.error('Configuration file %s does not exist', config_file)
-        sys.exit(1)
+        raise EnosFilePathError(
+            config_file, "Configuration file %s does not exist" % config_file)
 
     # Calls the provider and initialise resources
     provider = make_provider(env)
     config = load_config(env['config'],
                          provider.topology_to_resources,
                          provider.default_config())
     rsc, provider_net, eths = \
-        provider.init(config, CALL_PATH, kwargs['--force-deploy'])
+        provider.init(config, kwargs['--force-deploy'])
 
     env['rsc'] = rsc
     env['provider_net'] = provider_net
     env['eths'] = eths
 
-    logging.debug("Provides ressources: %s", env['rsc'])
-    logging.debug("Provides network information: %s", env['provider_net'])
-    logging.debug("Provides network interfaces: %s", env['eths'])
+    logging.debug("Provider ressources: %s", env['rsc'])
+    logging.debug("Provider network information: %s", env['provider_net'])
+    logging.debug("Provider network interfaces: %s", env['eths'])
 
     # Generates inventory for ansible/kolla
-    base_inventory = env['config']['inventory']
+    base_inventory = seekpath(env['config']['inventory'])
     inventory = os.path.join(env['resultdir'], 'multinode')
     generate_inventory(env['rsc'], base_inventory, inventory)
     logging.info('Generates inventory %s' % inventory)
 
     env['inventory'] = inventory
 
+    # Wait for resources to be invariable
     wait_ssh(env)
 
     # Set variables required by playbooks of the application
-    if 'ip' in env['config']['registry']:
-        registry_vip = env['config']['registry']['ip']
-    else:
-        registry_vip = pop_ip(env)
-
     env['config'].update({
         'vip':               pop_ip(env),
-        'registry_vip':      registry_vip,
+        'registry_vip':      env['config']['registry'].get('ip')or pop_ip(env),
```

Comment:
You got me! I did that to stay in the limit of 74 chars 😄, but you are right I will add a space.